### PR TITLE
Code simplification (mostly) background js

### PR DESCRIPTION
**Pull request by brunoais**
Here's part 2 of the cleanup and reorganizing.

The last two commits weren't easy to apply but all seems to be working, at first glance.

Check if all looks good and, as it's been, let me know if sth is wrong.

I thought this would take longer than it took, which is nice...

### CODE REVIEW

**Comment by brunoais** [comment]
Ugh... I don't know how to solve some of these conflicts without force-pushing...

File: hcaptcha.js [hunk 1]
```diff
@@ -1,176 +1,4 @@
...
-                }
-            }
-
-            return {job_id, clicks: null};
-        }
-    }
-
-
-    // const {Time, BG, NopeCHA} = await import(chrome.runtime.getURL('utils.js'));
-
```
Comment (brunoais) [inline hcaptcha.js:173]: Most of the above block was moved and split into `utils.mjs` & `content_scripts.js`

Comment (brunoais) [inline hcaptcha.js:173]: Most of the above block was moved and split into `utils.mjs` & `content_scripts.js`

File: hcaptcha.js [hunk 2]
```diff
@@ -224,32 +52,21 @@
...
-            '0435': 'e',
-            '0440': 'p',
-            '0441': 'c',
-            '0443': 'y',
-            '0455': 's',
-            '0456': 'i',
-            '0501': 'd',
-            '30fc': '一',
-            '571f': '士',
+            '0335': '-',
```
Comment (brunoais) [inline hcaptcha.js:69]: This came when I tried to merge with the master. I now know I did the process not so well but I don't know how to fix without a force-push.

File: recaptcha.js [hunk 3]
```diff
@@ -1,174 +1,4 @@
...
-                    break;
-                }
-            }
-
-            return {job_id, clicks: null};
-        }
-    }
-
-
-    // const {Time, BG, NopeCHA} = await import(chrome.runtime.getURL('utils.js'));
```
Comment (brunoais) [inline recaptcha.js:171]: Most of the above block was moved and split into `utils.mjs` & `content_scripts.js`

File: utils.js [hunk 4]
```diff
@@ -1,268 +1,15 @@
...
+/*******************************
 
+This file is automatically generated by build.py based on utils.mjs
+If you need something on utils.js, edit utils.mjs instead
 
-class Type {
-    static _string_constructor = 'string'.constructor;
-    static _array_constructor = [].constructor;
-    static _object_constructor = ({}).constructor;
+*********************************/
```
Comment (brunoais) [inline utils.js:6]: `utils.js` is now auto-generated from `utils.mjs`. It's the exact same code but in a non-module way.

I was unable to make it a single file while making it work in both modes. I also heavily dislike repeated code in chunks of this size.

File: version/chrome/api.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                    'resourceTypes': [
+                        'sub_frame',
+                        'script',
+                    ],
+                },
+            },
+        ],
+        removeRuleIds: [1],
+    });
+}
```
Comment (brunoais) [inline version/chrome/api.js:36]: api files provide the browser-specific api differences to background code in a browser-agnostic way.

I could have made some sort of API transformer which works with both but that would be making new API without actual need. Instead, I decided to go forward with complete copies on both sides which is much faster and it's very small, no issues.

File: version/chrome/manifest.json [hunk 6]
```diff
@@ -3,6 +3,9 @@
     "permissions": [
         "declarativeNetRequest"
     ],
-    "background": {"service_worker": "background.js"},
+    "background": {
+        "service_worker": "background.js",
+        "type": "module"
```
Comment (brunoais) [inline version/chrome/manifest.json:8]: Required so the `import` can be used

File: version/firefox/manifest.json [hunk 7]
```diff
@@ -6,6 +6,11 @@
         "<all_urls>"
     ],
     "background": {
-        "scripts": ["background.js"]
+        "page": "page.html"
```
Comment (brunoais) [inline version/firefox/manifest.json:9]: Allows loading `background.js` as a module in the background page.

**Comment by brunoais** [comment]
I think I got it worked out. What I needed was a specific merge strategy:
`git merge --no-ff --strategy=ours origin/main`

**Review by Jaewan-Yun — APPROVED** [review]

**Pull request merged**